acctz/acctz.proto

```diff
@@ -99,34 +99,30 @@ message SessionInfo {
 // Details of authentication - for gRPC or shell/vendor-CLI.
 message AuthDetail {
   // Identity string is used to identify the user that executed the
-  // command. For instance, it could be the spiffe-id in the case of
+  // command. For instance, it could be the SPIFFE-ID in the case of
   // gRPC or unix-style user-name in the case of shell/vendor-CLI.
   string identity = 1;
 
-  // Privilege level configured on the system.
-  uint32 privilege_level = 2;
```

Comment:
Privilege levels are custom to only some implementations and generally sit between the range of 0-15 (16 values) with a few system defined levels that carry similar behaviors leaving room for user-defined levels and privilege mapping.

Now, from this I would like request for comment on the following:
* Does this even need to be sent for every accounting record?  User permissions mappings to resources are abstract of this privilege level custom to only some implementations.  Others may use other techniques such as user classes/groups and that is not conveyed here
* If it becomes optional for implementations that do not support then we have an issue omitting the field as it would take on the assumption of the value 0 for protobuf (which can be mixed in with an actual value).  One option would be to always set something like 2^32-1 (4294967295) in such cases but that is also unnecessary overhead

Reply:
I was thinking about this separately.  priv-lvl is not universal; other platforms use user-defined names to identify roles, such as "operator".  As such, I think this should be a string, not an int.  It would be more versatile that way and could still hold a number.
It could be moved, but I think some parity with existing accounting is useful and do not consider this overhead.

Reply:
I think this goes back to the intentions and goals of gNSI - being a redefinition of AAA, it seems there is a mix of items here that are not covered in their (OpenConfig) YANG model or other gNSI service equivalents (authorization/authz/pathz) but maybe rather map back to existing per-vendor implementation behaviors.

A "user" is bound to a privilege-level internal to a given implementation but this is agnostic to the binding by way of gNSI (and what seems like for good reason).  Correct me if I'm wrong but this "binding" should be known, can be queryable/streamed or its own structure table (aka. AuthorizationPolicy) so does every accounting record need to repeat this?

The answer could very well be yes but to draw the analogy to the gNMI/YANG world, this is like separate views of the same data just organized differently and when looking across all OC projects in tandem - there is an explosion of data by way of modeling across various APIs to be implemented on an element with finite resources.

Just to elaborate on my concern of "overhead", its with respect to a view across the board on what is ultimately protocol design so every bit will count - this is by no means a field that will be the offender but my concern would currently be targeted at full payload reflections in either success/failure scenarios especially when dealing with very large config sets (Coupled with OpenConfig modeled expansions that are likely to magnify a data payload an order of magnitude from previous as items like apply-groups, macro expansions, etc.. are not used)

Reply:
If the 'user' here can change their privilege level during a session/connection I think you need to track that.
In one example. this is:

  ```login as me, enable to do RW things, logout```
  
if you only noted the role/priv-level/etc at the start/end you'd be missing any transitions.
'not an int' seems like a good idea though.

Reply:
And, because it can change and because accounting records are not necessarily received/retrieved while the user is still logged-in or even at the same priv-lvl, it might not be possible to retrieve the level in use at the time of an event.

Reply:
This field should just be a string.  Rename it to something generic, allowing it to cover the various implementations: privilege_level, user class, user role group, etc etc.  something like

// user's privilege level, user class, user group, task group, etc.
string group;

Reply:
I'll add this to PR #122